server/src/main/java/org/opensearch/indices/recovery/MultiFileWriter.java

```diff
@@ -156,7 +156,10 @@ private void innerWriteFileChunk(StoreFileMetadata fileMetadata, long position,
                 + temporaryFileName
                 + "] in "
                 + Arrays.toString(store.directory().listAll());
-            store.directory().sync(Collections.singleton(temporaryFileName));
+            // In the case of segment replication, we want files to be synced only when they're part of a greater incoming commit point
+            if (!store.indexSettings().isSegRepEnabled() || indexState.hasIncomingCommitPoint()) {
```

Comment:
from our [developer guide](https://github.com/opensearch-project/OpenSearch/blob/b0080eb8dfe905f15a2df92bd3a349222e0df1b3/DEVELOPER_GUIDE.md#java-language-formatting-guidelines), please use == false for negative boolean expressions.